include/aikido/planner/dart/ConfigurationToConfiguration.hpp

```diff
@@ -0,0 +1,75 @@
+#ifndef AIKIDO_PLANNER_DART_CONFIGURATIONTOCONFIGURATION_HPP_
+#define AIKIDO_PLANNER_DART_CONFIGURATIONTOCONFIGURATION_HPP_
+
+#include <dart/dart.hpp>
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/planner/Problem.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+
+namespace aikido {
+namespace planner {
+namespace dart {
+
+/// Planning problem to plan to a single goal configuration.
+class ConfigurationToConfiguration : public Problem
+{
+public:
+  /// Constructor. Note that this constructor takes the start state from the
+  /// current state of the passed MetaSkeleton.
+  ///
+  /// \param[in] stateSpace State space.
+  /// param[in] metaSkeleton MetaSkeleton that getStartState will return the
+  /// current state of when called.
+  /// \param[in] goalState Goal state to plan to.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  ConfigurationToConfiguration(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr stateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton,
+      const statespace::dart::MetaSkeletonStateSpace::State* goalState,
+      constraint::ConstTestablePtr constraint);
```

Comment:
Can we allow `nullptr` for constraint and make it default to `Satisfied` if `nullptr` is passed?

Reply:
It looks like the default `Problem` class handles this:

https://github.com/personalrobotics/aikido/blob/fe09cbb5d2f57a3366eb1cc4a8ec932f1436be0c/src/planner/Problem.cpp#L17

Reply:
@sniyaz That's true. I think all we need to do here is just to pass `nullptr` as the default value to `constraint` so that the default `Problem` do the job.